### DISCUSSION TITLE
(Soft robotics) how can I run a FEA simulation of a soft actuator that uses a network of ropes (strings)?

### DISCUSSION THREAD

**Discussion by IKEAmeatballs039** [sofa-framework/sofa#4959] [SofaPython3] [answered]
I am completely new here, so I’m very sorry if this is the wrong place to ask such a question.

i want to create a fea simulation of a soft robot design that is made of linear soft actuators hooked to a looped piece of string (the string limits how much each actuator can move)

Does sofa have assets that can be used as strings or non elastic ropes? Is it possible to configure how the string loop attatches itself to the actuators and how it crosses over itself?

Again, I’m new here and this is my first time posting here, so I’m really sorry if I posted this in the wrong place.

**Comment by bakpaul** [comment]
Hello ! 

What you are asking is totally doable currently with SOFA ! We have an available plugin suit for soft robotics : [SoftRobots](https://github.com/SofaDefrost/SoftRobots) and [SoftRobots.Inverse ](https://github.com/SofaDefrost/SoftRobots.Inverse), the first one enabling forward simulation of soft robots in SOFA and the second one enabling to control them. 

Using those plugins you'll be able to use linear cable actuator, pressure chamber actuators and others. Take a look at the available example (I guess that the trunk example might be the best one for you). 

Know that we have a specific training course on soft robotics with SOFA. When you are sure that this is really the solution you need, you can enroll for a training course here : https://www.sofa-framework.org/sofa-events/sofa-training-sessions/ 

See you !

**Reply by IKEAmeatballs039** [reply]
Thank you for your kind reply!

I plan on using the softrobots plugin, but it didn't specify if it was possible to use cables or strings in a complex way (things with knots for example) Do you know if this is possible? If so, how can I specify how a cable is knotted or positioned?

**Reply by bakpaul** [reply, marked as answer]
Cable simulation, outside of being used as actuators for robots (e.g. if you want to manipulate cables with a robot)  is a totally different simulation. We have several plugins allowing to simulate cables : [BeamAdapter](https://github.com/sofa-framework/BeamAdapter) and [Cosserat](https://github.com/SofaDefrost/Cosserat), one using the Timoshenko formulation and the second one the Cosserat theory. You can checkout the examples to see their capabilities !

**Reply by IKEAmeatballs039** [reply]
Thank you so much! Ill check them out!
